libmamba/include/mamba/core/util_string.hpp

```diff
@@ -112,86 +114,146 @@ namespace mamba
                 left += right;
             }
         };
-    }
 
-    template <class S, class CharType, class Joiner = details::PlusEqual>
-    auto join(const CharType* sep, const S& container, Joiner joiner = details::PlusEqual{}) ->
-        typename S::value_type
-    {
-        if (container.empty())
-            return {};
-        auto result = container[0];
-        for (std::size_t i = 1; i < container.size(); ++i)
+        template <class T, class = void>
+        struct has_reserve : std::false_type
+        {
+        };
+        template <class T>
+        struct has_reserve<T, std::void_t<decltype(std::declval<T>().reserve(std::size_t()))>>
+            : std::true_type
+        {
+        };
+        template <typename T>
+        inline constexpr bool has_reserve_v = has_reserve<T>::value;
+
+        std::size_t size(const char* s);
+        std::size_t size(const wchar_t* s);
+        std::size_t size(const char c);
+        template <class T>
+        inline std::size_t size(T const& s)
```

Comment:
The `inline` here is unnecessary as any template function is implicitly `inline`.

By the way, why not just reuse `std::size` here?

Reply:
Similarly it's for the generic code. This function could use `std::size` but using it directly in the generic code would require some sort of ADL.